### PR TITLE
Step 1 in fixing the Timer/Thread issues on shutdown

### PR DESCRIPTION
**Pull request by ZappoMan**
* Cleaned up some old dead code that wasn't in use

* implemented support for LimitedNodeList to move itself back to the main thread on shutdown. Basically in the Interface application we were moving the NodeList to a different thread, and that thread shuts down before we delete the node list instance. So when QObject is cleaning up, it displays this warning (and likely causes other unknown problems).

* Fixes a crash on shutdown related to MyAvatar

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5867f40bc7a8663b2ffdd6b41530d84dde1cfa76 available at: https://gist.github.com/4ddd7ee8ef580fb345f9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4421.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4421.dmg

**Comment by birarda** [comment]
Ran into similar problems with the AudioClient.

I think the fix for this is actually to make sure that the LNL uses a custom deleter with the DependencyManager, and then let it clean itself and it's thread up when the DependencyManager goes to get rid of it. Otherwise the DependencyManager cleanup for LNL (and anything by default) is just `delete` and that's what causes these issues

**Comment by birarda** [comment]
Brought this up with @Atlante45 previously ... ideally I think we want the DependencyManager deleter for all QObjects to be `deleteLater` instead of `delete object` or I would expect to see this message for any QObject on a thread with QTimer children.

**Comment by AndrewMeadows** [comment]
I'm assigning this to @birarda since it sounds like he knows more about this than I do.

**Comment by ZappoMan** [comment]
This is not an issue of the deleter.. this is an issue that you must move the NLN back to the application thread before it gets deleted... actually you have to do it while on the thread you moved it to... because you can "pull" objects to a thread, you can only "push" objects to a thread. This is explained in greater detail in the Qt docs.

This PR also fixes a crash in AvatarManager->MyAvatar-> ~Resource

**Comment by birarda** [comment]
It is possible to have the LNL and the thread it is on be cleaned up when the Application is going down, without pushing it from the LNL thread to the Application thread. This is what we do for the AudioClient.

File: libraries/networking/src/DomainHandler.cpp [hunk 1]
```diff
@@ -50,12 +49,6 @@ void DomainHandler::clearConnectionInfo() {
     }
     
     setIsConnected(false);
-
```
Comment (ZappoMan) [inline libraries/networking/src/DomainHandler.cpp]: note: all this is dead code. _handshakeTimer is never set.

File: assignment-client/src/audio/AudioMixer.cpp [hunk 2]
```diff
@@ -37,7 +37,6 @@
 #include <QtCore/QJsonObject>
 #include <QtCore/QJsonValue>
 #include <QtCore/QThread>
-#include <QtCore/QTimer>
```
Comment (ZappoMan) [inline assignment-client/src/audio/AudioMixer.cpp]: note: dead code QTimer never used here

File: assignment-client/src/Agent.cpp [hunk 3]
```diff
@@ -12,7 +12,6 @@
 #include <QtCore/QCoreApplication>
 #include <QtCore/QEventLoop>
 #include <QtCore/QStandardPaths>
-#include <QtCore/QTimer>
```
Comment (ZappoMan) [inline assignment-client/src/Agent.cpp]: note: dead code QTimer never used here

File: interface/src/Application.cpp [hunk 4]
```diff
@@ -612,7 +612,8 @@ Application::~Application() {
     
     // stop the glWidget frame timer so it doesn't call paintGL
     _glWidget->stopFrameTimer();
-    
+
+    DependencyManager::destroy<AvatarManager>();
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: this fixes the crash on shutdown related to MyAvatar attempting to access a texture... if we force the cleanup order to destroy the AvatarManager before TextureCache we fix this crash.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d03aa83263db5c1dd4e305542092b32c33698a1c available at: https://gist.github.com/4ddd7ee8ef580fb345f9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4421.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4421.dmg

**Pull request closed without merging**